R/assert_age_formatted.R

```diff
@@ -0,0 +1,122 @@
+#' Are the age related columns formatted correctly?
+#'
+#' Checks to see that each set of age groups for each combination of the id
+#' columns is formatted correctly. This includes checking that:
+#' * the age groups are contiguous.
+#' * there are no overlapping age groups.
+#' * the youngest age group starts at age `first_age_start`.
+#' * the oldest age group ends at `terminal_age_end`.
+#'
+#' If `assert_age_formatted_dt` throws an error indicating the dataset is not
+#' formatted correctly, you can use `identify_age_misformatted_dt` to see the rows
+#' of the dataset that are problematic.
+#'
+#' @param dt input data.table to check.
+#' * must contain columns specified in `id_cols`.
+#' * must include numeric columns called 'age_start' and 'age_end' that contain
+#' no missing values.
+#' * each combination of `id_cols` must uniquely identify each row.
+#' @param id_cols character vector of id columns that uniquely identify each row
+#' of `dt`.
+#' * must include 'age_start' and 'age_end'.
+#' @param first_age_start the numeric 'age_start' value for the first age group.
+#' @inheritParams calculate_age_end
+#'
+#' @return
+#' * `identify_age_misformatted_dt` returns a data.table with problematic age
+#' groups only, includes a column 'error' that identifies 'missing' or
+#' 'overlapping' age ranges.
+#' * `assert_age_formatted_dt` returns nothing but throws an error if
+#' `identify_age_misformatted_dt` returns a non empty data.table.
+#'
+#' @export
```

Comment:
does it matter where the export is? I've seen it at the end of the header and in the middle, and I'm curious if it makes a difference and/or if there is a convention.

Reply:
Doesn't matter I'm pretty sure. Not sure if there is a convention either. 

I think usually we have title, description, parameters at the beginning. Then examples at the end.

@spencerpease thoughts?